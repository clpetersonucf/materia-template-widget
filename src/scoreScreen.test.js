import React from 'react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'

const mockDomRender = jest.fn()

// jest.mock('react-dom', () => ({ReactDom: {render: mockDomRender}}), {virtual: true})
// jest.mock('./react-render', () => mockDomRender, {virtual: true})

describe('ScoreScreen', function() {
	beforeEach(() => {
		jest.resetAllMocks()
		jest.resetModules()
		jest.unmock('react-dom')
		global.Materia = {
			ScoreCore: {
				start: jest.fn(),
				update: jest.fn(),
				hideResultsTable: jest.fn()
			}
		}
	});

	test('ScoreScreen registers callbacks', () => {
		require('./scoreScreen')
		jest.mock('react-dom', () => {
			jest.fn();
		})
		expect(Materia.ScoreCore.start).toHaveBeenCalledTimes(1)
		const callbacks = Materia.ScoreCore.start.mock.calls[0][0]
		expect(callbacks).toHaveProperty('start', expect.any(Function))
		expect(callbacks).toHaveProperty('update', expect.any(Function))
	});

	test('start renders as expected', () => {
		require('./scoreScreen')
		jest.mock('react-dom', () => ({
			render: mockDomRender
		}))

		const start = Materia.ScoreCore.start.mock.calls[0][0].start

		const scoreTable = [
			{
				data: [
					"Mock Question 1",
					"Mock Correct Answer 1",
					"Mock Given Answer 1"
				],
				score: 100,
				symbol: "%"
			},
			{
				data: [
					"Mock Question 2",
					"Mock Correct Answer 2",
					"Mock Given Answer 2"
				],
				score: 0,
				symbol: " points"
			}
		]

		expect(mockDomRender).toHaveBeenCalledTimes(0)
		start({name: 'mock-name'}, {}, scoreTable, false, 1)
		expect(mockDomRender).toHaveBeenCalledTimes(1)
		expect(mockDomRender.mock.calls[0][0]).toMatchSnapshot()
	});

	test('update renders as expected', () => {
		require('./scoreScreen')
		jest.mock('react-dom', () => ({
			render: mockDomRender
		}))

		const update = Materia.ScoreCore.start.mock.calls[0][0].update

		const scoreTable = [
			{
				data: [
					"Mock Question 1",
					"Mock Correct Answer 1",
					"Mock Given Answer 1"
				],
				score: 100,
				symbol: "%"
			},
			{
				data: [
					"Mock Question 2",
					"Mock Correct Answer 2",
					"Mock Given Answer 2"
				],
				score: 0,
				symbol: " points"
			}
		]

		expect(mockDomRender).toHaveBeenCalledTimes(0)
		update({}, scoreTable)
		expect(mockDomRender).toHaveBeenCalledTimes(1)
		expect(mockDomRender.mock.calls[0][0]).toMatchSnapshot()
	});

	test('ScoreScreen renders empty table', () => {
		const ScoreScreenApp = require('./scoreScreen').default
		const props = {
			title: 'mock-title',
			scoreTable: []
		}

		const component = renderer.create(<ScoreScreenApp {... props}/>);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('ScoreScreen renders title and table', () => {
		const ScoreScreenApp = require('./scoreScreen').default
		const props = {
			title: 'mock-title',
			scoreTable: [
				{
					data: [
						"Mock Question 1",
						"Mock Correct Answer 1",
						"Mock Given Answer 1"
					],
					score: 100,
					symbol: "%"
				},
				{
					data: [
						"Mock Question 2",
						"Mock Correct Answer 2",
						"Mock Given Answer 2"
					],
					score: 0,
					symbol: " points"
				}
			]
		}

		const component = renderer.create(<ScoreScreenApp {... props}/>);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
})
