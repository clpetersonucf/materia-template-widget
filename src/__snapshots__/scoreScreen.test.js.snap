// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreScreen ScoreScreen renders empty table 1`] = `
<div>
  <div
    id="title"
  >
    mock-title
  </div>
  <div>
    Score Table
  </div>
</div>
`;

exports[`ScoreScreen ScoreScreen renders title and table 1`] = `
<div>
  <div
    id="title"
  >
    mock-title
  </div>
  <div>
    Score Table
  </div>
  <div>
    <div>
      Question
    </div>
    <div>
      Mock Question 1
    </div>
    <div>
      Score
    </div>
    <div>
      100
      %
    </div>
    <div>
      Correct Answer
    </div>
    <div>
      Mock Correct Answer 1
    </div>
    <div>
      Given Answer
    </div>
    <div>
      Mock Given Answer 1
    </div>
  </div>
  <div>
    <div>
      Question
    </div>
    <div>
      Mock Question 2
    </div>
    <div>
      Score
    </div>
    <div>
      0
       points
    </div>
    <div>
      Correct Answer
    </div>
    <div>
      Mock Correct Answer 2
    </div>
    <div>
      Given Answer
    </div>
    <div>
      Mock Given Answer 2
    </div>
  </div>
</div>
`;

exports[`ScoreScreen start renders as expected 1`] = `
<ScoreScreenApp
  qset={Object {}}
  scoreTable={
    Array [
      Object {
        "data": Array [
          "Mock Question 1",
          "Mock Correct Answer 1",
          "Mock Given Answer 1",
        ],
        "score": 100,
        "symbol": "%",
      },
      Object {
        "data": Array [
          "Mock Question 2",
          "Mock Correct Answer 2",
          "Mock Given Answer 2",
        ],
        "score": 0,
        "symbol": " points",
      },
    ]
  }
  title="mock-name"
/>
`;

exports[`ScoreScreen update renders as expected 1`] = `
<ScoreScreenApp
  qset={Object {}}
  scoreTable={
    Array [
      Object {
        "data": Array [
          "Mock Question 1",
          "Mock Correct Answer 1",
          "Mock Given Answer 1",
        ],
        "score": 100,
        "symbol": "%",
      },
      Object {
        "data": Array [
          "Mock Question 2",
          "Mock Correct Answer 2",
          "Mock Given Answer 2",
        ],
        "score": 0,
        "symbol": " points",
      },
    ]
  }
  title=""
/>
`;
